### PR TITLE
Sound is playing block added to the volume category for music blocks

### PR DESCRIPTION
**Pull request by srietkerk**
When using the `on loud sound` block, if you are also having audio play through the speakers, it often triggers the `loud sound` on hardware. The new `is sound playing` block will help in situations like this. For users who want this outcome of playing sounds and triggering `on loud sound` this will not effect their experience. However, if users don't want that experience, they can use this block with some logic in `on loud sound`, and the experience will better align with what is desired.

What it looks like in the bin:
image
 

Example:
image

**Dependent on** https://github.com/microsoft/pxt/pull/9541

Closes https://github.com/microsoft/pxt-microbit/issues/3777
Upload target: https://makecode.microbit.org/app/ec7f196399d22b881359b4b47f07da2d2b50783c-242591ac95

### CODE REVIEW

File: sim/state/music.ts [hunk 1]
```diff
@@ -11,4 +11,10 @@ namespace pxsim.music {
     export function setSilenceLevel(level: number) { 
         // ignore in v1,v2
     }
+
+    export function isSoundPlaying(): boolean {
+        const audioActive = pxsim.AudioContextManager.isAudioElementActive();
+        const soundExpressionPlaying = pxsim.codal.music.isSoundExpPlaying();
+        return audioActive || soundExpressionPlaying;
```
Comment (jwunderl) [inline sim/state/music.ts:18]: Does this also need to check for whether a microphone recording is playing?

**Review by jwunderl — APPROVED** [review]
LGTM, though I think it might be missing a check for whether the microphone recording is active?

File: sim/state/music.ts (hunk 1 above)
Comment (srietkerk) [reply sim/state/music.ts:18]: Yes. Thank you for catching this. I remembered at some point today and then completely forgot.

**Review by kimprice — APPROVED** [review]
Looks good!

**Review by thsparks — APPROVED** [review]

File: libs/core/music.cpp [hunk 2]
```diff
@@ -63,6 +63,21 @@ void setBuiltInSpeakerEnabled(bool enabled) {
...
+*/
+//% blockId=music_sound_is_playing block="sound is playing"
+//% group="Volume"
+//% help=music/volume
+//% weight=0
+bool isSoundPlaying() {
+#if MICROBIT_CODAL
+    uBit.audio.isPlaying();
+#else
+    target_panic(PANIC_VARIANT_NOT_SUPPORTED);
```
Comment (abchatra) [inline libs/core/music.cpp:77]: I realized we don't support v1. We should probably group this block with v2 only blocks.

**Pull request merged**